src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/DeliveryRequestResolver.java

```diff
@@ -48,12 +47,15 @@ public boolean canResolve(
 
         final Colony colony = (Colony) manager.getColony();
         final CitizenData freeDeliveryMan = colony.getCitizenManager().getCitizens()
-                                        .stream()
-                                        .filter(c -> c.getCitizenEntity() != null
-                                                && requestToCheck.getRequest().getTarget().isReachableFromLocation(c.getCitizenEntity().getLocation()))
-                                        .filter(c -> c.getJob() instanceof JobDeliveryman)
-                                        .findFirst()
-                                        .orElse(null);
+                                              .stream()
+                                              .filter(citizenData -> citizenData.getCitizenEntity()
+                                                                       .map(entityCitizen -> requestToCheck.getRequest()
+                                                                                               .getTarget()
+                                                                                               .isReachableFromLocation(entityCitizen.getLocation()))
+                                                                       .orElse(false))
+                                              .filter(c -> c.getJob() instanceof JobDeliveryman)
+                                              .findFirst()
+                                              .orElse(null);
 
         if (freeDeliveryMan == null)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)